pkg/acme/client/middleware/logger.go

```diff
@@ -38,57 +43,101 @@ var _ client.Interface = &Logger{}
 
 func (l *Logger) AuthorizeOrder(ctx context.Context, id []acme.AuthzID, opt ...acme.OrderOption) (*acme.Order, error) {
 	klog.Infof("Calling CreateOrder")
-	return l.baseCl.AuthorizeOrder(ctx, id, opt...)
+
+	acmectx, cancel := l.acmeContext(ctx)
+	defer cancel()
```

Comment:
I'm not sure if overloading this `Logger` struct to also handle contexts is really the best place for this - this whole structure is likely to go away at some point as it's not a _particularly_ helpful logger (it doesn't use `logr` meaning you cannot associate log messages from it with different resources being synced).

Given this has merged already I think it's acceptable, but this whole package needs some TLC I think 😬